### ISSUE TITLE
Unrecognized target bytecode: '8'

### ISSUE THREAD

**Issue by xiumin1993** [groovy/GMavenPlus#223]
I'm using    gmavenplus-plugin  1.13.1
when I setting maven
```
<maven.compiler.source>8</maven.compiler.source>
<maven.compiler.target>8</maven.compiler.target>
```
cmd
```
mvn test
```
get  error  
```
Caused by: java.lang.IllegalArgumentException: Unrecognized target bytecode: '8'.
    at org.codehaus.gmavenplus.mojo.AbstractCompileMojo.verifyGroovyVersionSupportsTargetBytecode (AbstractCompileMojo.java:544)
    at org.codehaus.gmavenplus.mojo.AbstractCompileMojo.doCompile (AbstractCompileMojo.java:314)
    at org.codehaus.gmavenplus.mojo.CompileTestsMojo.execute (CompileTestsMojo.java:77)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)

```

Maybe should add java version 6 to 8 in this function
```
  /**
     * Throws an exception if targetBytecode is not supported with this version of Groovy. That is, when Groovy added
     * the option to org.codehaus.groovy.control.CompilerConfiguration and used it in
     * org.codehaus.groovy.classgen.asm.WriterController.
     */
    protected void verifyGroovyVersionSupportsTargetBytecode() {
        if ("18".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_4_0_0_BETA1)) {
                throw new IllegalArgumentException("Target bytecode 18 requires Groovy " + GROOVY_4_0_0_BETA1 + " or newer.");
            }
        } else if ("17".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_3_0_8) || (groovyAtLeast(GROOVY_4_0_0_ALPHA1) && groovyOlderThan(GROOVY_4_0_0_ALPHA3))) {
                throw new IllegalArgumentException("Target bytecode 17 requires Groovy " + GROOVY_3_0_8 + "/" + GROOVY_4_0_0_ALPHA3 + " or newer.");
            }
        } else if ("16".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_3_0_6)) {
                throw new IllegalArgumentException("Target bytecode 16 requires Groovy " + GROOVY_3_0_6 + " or newer.");
            }
        } else if ("15".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_3_0_3)) {
                throw new IllegalArgumentException("Target bytecode 15 requires Groovy " + GROOVY_3_0_3 + " or newer.");
            }
        } else if ("14".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_3_0_0_BETA2)) {
                throw new IllegalArgumentException("Target bytecode 14 requires Groovy " + GROOVY_3_0_0_BETA2 + " or newer.");
            }
        } else if ("13".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_2_5_7) || (groovyAtLeast(GROOVY_2_6_0_ALPHA1) && groovyOlderThan(GROOVY_3_0_0_BETA1))) {
                throw new IllegalArgumentException("Target bytecode 13 requires Groovy " + GROOVY_2_5_7 + "/" + GROOVY_3_0_0_BETA1 + " or newer. No 2.6 version is supported.");
            }
        } else if ("12".equals(targetBytecode) || "11".equals(targetBytecode) || "10".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_2_5_3) || (groovyAtLeast(GROOVY_2_6_0_ALPHA1) && groovyOlderThan(GROOVY_3_0_0_ALPHA4))) {
                throw new IllegalArgumentException("Target bytecode 10, 11, or 12 requires Groovy " + GROOVY_2_5_3 + "/" + GROOVY_3_0_0_ALPHA4 + " or newer. No 2.6 version is supported.");
            }
        } else if ("9".equals(targetBytecode)) {
            if (!isGroovyIndy() && (groovyOlderThan(GROOVY_2_5_3)
                    || (groovyAtLeast(GROOVY_2_6_0_ALPHA1) && groovyOlderThan(GROOVY_2_6_0_ALPHA4))
                    || (groovyAtLeast(GROOVY_3_0_0_ALPHA1) && groovyOlderThan(GROOVY_3_0_0_ALPHA2)))) {
                throw new IllegalArgumentException("Target bytecode 9 requires Groovy " + GROOVY_2_5_3 + "/" + GROOVY_2_6_0_ALPHA4 + "/" + GROOVY_3_0_0_ALPHA2 + " or newer.");
            } else if (isGroovyIndy() && (groovyOlderThan(GROOVY_2_5_3) || (groovyAtLeast(GROOVY_2_6_0_ALPHA1) && groovyOlderThan(GROOVY_3_0_0_ALPHA4)))) {
                throw new IllegalArgumentException("Target bytecode 9 with invokedynamic requires Groovy " + GROOVY_2_5_3 + "/" + GROOVY_3_0_0_ALPHA4 + " or newer. No 2.6 version is supported.");
            }
        } else if ("1.8".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_2_3_3)) {
                throw new IllegalArgumentException("Target bytecode 1.8 requires Groovy " + GROOVY_2_3_3 + " or newer.");
            }
        } else if ("1.7".equals(targetBytecode) || "1.6".equals(targetBytecode)) {
            if (groovyOlderThan(GROOVY_2_1_3)) {
                throw new IllegalArgumentException("Target bytecode 1.6 and 1.7 require Groovy " + GROOVY_2_1_3 + " or newer.");
            }
        } else if (!"1.5".equals(targetBytecode) && !"1.4".equals(targetBytecode)) {
            throw new IllegalArgumentException("Unrecognized target bytecode: '" + targetBytecode + "'.");
        }
    }
```

**Comment by keeganwitt** [comment]
It's also worth noting you can use `skipBytecodeCheck` to skip this verification.

**Comment by keeganwitt** [comment]
I can't find the source code for it, but it looks like the synonyms are what you'd expect (https://docs.oracle.com/en/java/javase/11/tools/javac.html#GUID-AEEC9F07-CB49-4E96-8BC7-BCC2C7F725C9). I'll add these.

In the mean time, it works for me with the below XML
```xml
<properties>
  <skipBytecodeCheck>true</skipBytecodeCheck>
</properties>
```

**Comment by keeganwitt** [comment]
Actually, I'm remembering why I implemented it this way now. It's because while javac accepts both `1.8` and `8`, Groovy does not. See https://github.com/apache/groovy/blob/38c85f371128b6bc46ba952228640d433b1c799e/src/main/java/org/codehaus/groovy/control/CompilerConfiguration.java

I suggest either
```xml
<properties>
  <maven.compiler.source>1.8</maven.compiler.source>
  <maven.compiler.target>1.8</maven.compiler.target>
</properties>
```
Since this would be equivalent and works for both javac and groovyc, or
```xml
<targetBytecode>1.8</targetBytecode>
```
The second of these can be placed in `<properties>` or in the `<configuration>` of GMavenPlus or one of its `<execution>`s.

**Comment by xiumin1993** [comment]
Thanks for your reply&nbsp; &nbsp;,You guys do the great things
I made a mistake&nbsp; setting&nbsp;&nbsp;skipBytecodeCheck=false  
you're rigth,it does work










------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "groovy/GMavenPlus"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年7月1日(星期五) 晚上11:33
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [groovy/GMavenPlus] Unrecognized target bytecode: '8' (Issue #223)





 
I can't find the source code for it, but it looks like the synonyms are what you'd expect (https://docs.oracle.com/en/java/javase/11/tools/javac.html#GUID-AEEC9F07-CB49-4E96-8BC7-BCC2C7F725C9). I'll add these.
 
In the mean time, it works for me with the below XML
 <properties&gt;   <skipBytecodeCheck&gt;true</skipBytecodeCheck&gt; </properties&gt;

**Comment by xiumin1993** [comment]
yeah，After check plugin source code ，I already change to this



发自我的iPhone


------------------ Original ------------------
From: Keegan Witt ***@***.***&gt;
Date: Fri,Jul 1,2022 11:45 PM
To: groovy/GMavenPlus ***@***.***&gt;
Cc: xiumin1993 ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [groovy/GMavenPlus] Unrecognized target bytecode: '8' (Issue#223)





 
Actually, I'm remembering why I implemented it this way now. It's because while javac accepts both 1.8 and 8, Groovy does not. See https://github.com/apache/groovy/blob/38c85f371128b6bc46ba952228640d433b1c799e/src/main/java/org/codehaus/groovy/control/CompilerConfiguration.java
 
I suggest either
 <properties&gt;   <maven.compiler.source&gt;1.8</maven.compiler.source&gt;   <maven.compiler.target&gt;1.8</maven.compiler.target&gt; </properties&gt;
 
Since this would be equivalent and works for both javac and groovyc, or
 <targetBytecode&gt;1.8</targetBytecode&gt;
 
The second of these can be placed in <properties&gt; or in the <configuration&gt; of GMavenPlus or one of its <execution&gt;s.

**Comment by keeganwitt** [comment]
I'm thinking about whether to translate the synonym to make this easier. I want to be careful I'm not breaking something else I haven't thought of.

https://github.com/groovy/GMavenPlus/pull/224

**Comment by keeganwitt** [comment]
I was more worried about if there was other JDK vendors (like Azul Zulu, Amazon Corretto, Alibaba Dragonwell, IBM, etc) that have other strings they accept that I haven't accounted for. I'm pretty confident this will work with Eclipse Adoptium and Oracle JDK.

**Comment by keeganwitt** [comment]
I released 2.0.0-SNAPSHOT with this change. As an additional verification, it'd be helpful to confirm this resolves your issue. You can get that from here

```xml
<repository>
  <id>sonatype-oss-snapshots</id>
  <url>https://oss.sonatype.org/content/repositories/snapshots</url>
  <releases>
    <enabled>false</enabled>
  </releases>
  <snapshots>
    <enabled>true</enabled>
    <updatePolicy>always</updatePolicy>
    <checksumPolicy>warn</checksumPolicy>
  </snapshots>
</repository>
```

**Comment by xiumin1993** [comment]
I think the best solustion is add a note on the exception message that they can use skipBytecodeCheck=true to skip this check when they get the same exception with me, also clearly descripe the exception message
“Unrecognized target bytecode: '8'”  not clear enough

发件人: Keegan Witt
发送时间: 2022年7月2日 1:41
收件人: groovy/GMavenPlus
抄送: xiumin1993; Author
主题: Re: [groovy/GMavenPlus] Unrecognized target bytecode: '8' (Issue#223)

I suppose if more cases come up, we can deal with them at that time. This seems reasonable to proceed with for now.
